src/routes/popup/agents/components/ao-yield/agent-info.tsx

```diff
@@ -0,0 +1,266 @@
+import styled from "styled-components";
+import { Section, Text, ListItem, Loading, Button } from "@arconnect/components-rebrand";
+import browser from "webextension-polyfill";
+import HeadV2 from "~components/popup/HeadV2";
+import { Flex } from "~components/common/Flex";
+import { PropertyName, PropertyValue, TransactionProperty } from "~routes/popup/transaction/[id]";
+import { useMemo, useState } from "react";
+import dayjs from "dayjs";
+import { useAOYieldAgent, useAOYieldAgentInfo } from "~utils/agents/hooks";
+import { Divider } from "~components/Divider";
+import { RemoveButton } from "~routes/popup/settings/wallets/[address]";
+import { SvgImageWithBackground } from "../SvgImage";
+import aoLogo from "url:/assets/ecosystem/ao-logo.svg";
+import { Settings01 } from "@untitled-ui/icons-react";
+import { PopupPaths } from "~wallets/router/popup/popup.routes";
+import { useLocation } from "~wallets/router/router.utils";
+import { AgentCancelModal } from "./AgentCancelModal";
+import type { WanderRoutePath } from "~wallets/router/router.types";
+import {
+  assets,
+  formatTokenQuantity,
+  getAOYieldAgents,
+  getStatusColor,
+  getStatusText,
+  setAOYieldAgents,
+} from "~utils/agents/utils";
+import type { MintingStatus } from "~utils/agents/types";
+import { useAsyncEffect } from "~utils/react/useAsyncEffect";
+import { getActiveAddress } from "~wallets";
+import { EventType, trackEvent } from "~utils/analytics";
+
+interface AgentInfoProps {
+  agentId: string;
+  headerTitle: string;
+  mintingStatus?: MintingStatus;
+  isHistory?: boolean;
+}
+
+export function AgentInfo({ agentId, headerTitle, mintingStatus, isHistory = false }: AgentInfoProps) {
+  const [showCancelModal, setShowCancelModal] = useState(false);
+  const agent = useAOYieldAgent(agentId);
+  const { data: agentInfo } = useAOYieldAgentInfo(agentId);
+  const { navigate } = useLocation();
+
+  const asset = useMemo(() => {
+    if (!agent) return null;
+    return assets.find((asset) => asset.id === agent.tokenOut);
+  }, [agent]);
+
+  const agentDetails = useMemo(() => {
+    if (!agent || !asset) return [];
```

Comment:
Very similar logic inside `src/routes/popup/agents/ao-yield/confirm-agent.tsx`. Maybe this could be extracted to a util function or hook?